pkg/utils/storage.go

```
@@ -133,3 +136,186 @@ func AddModelPvcMount(modelUri, containerName string, readOnly bool, podSpec *co
+		},
+		TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
+	}
+
+	return modelContainer
+}
+
+// ConfigureModelcarToContainer configures the OCI image specified in modelUri as a modelcar to the
+// specified target container of a given PodSpec. The configuration includes:
+//   - Adding an environment variable `async` to indicate to the runtime that the model directory may not be available immediately.
+//   - Setting the user ID for the target container, if specified in storageConfig.
+//   - Adding a modelcar and and init containers (for pre-fetching the model) if not already present.
```

golangci-lint (1.64.8)

Duplicate words (and) found

(dupword)